docker/Dockerfile.petsc

```diff
@@ -0,0 +1,121 @@
+# DockerFile for an environment into which firedrake can be installed. Based off
+# https://github.com/firedrakeproject/firedrake/blob/master/docker/Dockerfile.env
+
+FROM ubuntu:24.04
+
+# Update and install required packages for Firedrake
+USER root
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN apt-get update \
+    && apt-get -y dist-upgrade \
+    && DEBIAN_FRONTEND=noninteractive apt-get -y install tzdata \
+    && apt-get -y install curl vim docker.io openssh-client build-essential autoconf automake \
+        cmake gfortran git libopenblas-serial-dev libtool python3-dev python3-pip python3-tk \
+        python3-venv python3-requests zlib1g-dev libboost-dev sudo gmsh bison flex ninja-build \
+        libocct-ocaf-dev libocct-data-exchange-dev swig graphviz libcurl4-openssl-dev libxml2-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+# Use a more sane locale
+ENV LC_ALL C.UTF-8
+
+# Change the `ubuntu` user to `firedrake`
+# and ensure that we do not run as root on self-hosted systems
+RUN usermod -d /home/firedrake -m ubuntu \
+    && usermod -l firedrake ubuntu \
+    && groupmod -n firedrake ubuntu \
+    && usermod -aG sudo firedrake \
+    && echo "firedrake:docker" | chpasswd \
+    && echo "firedrake ALL=(ALL) NOPASSWD: ALL" >>/etc/sudoers \
+    && ldconfig
+
+USER firedrake
+WORKDIR /home/firedrake
+
+# Fetch PETSc
+RUN git clone https://github.com/firedrakeproject/petsc.git
+
+# Build MPICH manually because we don't want PETSc to build it twice
+RUN bash -c 'export PACKAGES=/home/firedrake/petsc/packages; \
+    cd petsc; \
+    ./configure \
+        --COPTFLAGS=-O3 -march=native -mtune=native \
+        --CXXOPTFLAGS=-O3 -march=native -mtune=native \
+        --FOPTFLAGS=-O3 -march=native -mtune=native \
+        --with-c2html=0 \
+        --with-debugging=0 \
+        --with-fortran-bindings=0 \
+        --with-make-np=12 \
+        --with-shared-libraries=1 \
+        --with-zlib \
```

Comment:
```
***** WARNING *****
Using version 1.3 of package ZLIB, PETSc is tested with 1.3. Suggest using
--download-zlib for a compatible zlib
```

Reply:
Let's leave it for now. I opened issue #41.